lambda-buffers-frontend/app/LambdaBuffers/Frontend/Cli/Compile.hs

```diff
@@ -1,15 +1,27 @@
 module LambdaBuffers.Frontend.Cli.Compile (CompileOpts (..), compile) where
 
 import Control.Lens (makeLenses, (^.))
-import Data.Map qualified as Map
+import Data.ByteString qualified as BS
+import Data.Maybe (fromMaybe)
+import Data.ProtoLens.Encoding qualified as Pb
+import Data.ProtoLens.TextFormat qualified as PbText
+import Data.Text.Lazy qualified as Text
+import Data.Text.Lazy.IO qualified as Text
 import LambdaBuffers.Frontend (runFrontend)
 import LambdaBuffers.Frontend.PPrint ()
-import Prettyprinter (Pretty (pretty))
-import Proto.Compiler ()
+import LambdaBuffers.Frontend.ToProto (toCompilerInput)
+import Proto.Compiler (CompilerInput)
+import System.FilePath ((<.>), (</>))
+import System.FilePath.Lens (extension)
+import System.IO.Temp (withSystemTempDirectory)
+import System.Process (callProcess)
 
 data CompileOpts = CompileOpts
   { _importPaths :: [FilePath]
   , _moduleFilepath :: FilePath
+  , _compilerCliFilepath :: FilePath
+  , _debug :: Bool
+  , _workingDir :: Maybe FilePath
   }
   deriving stock (Eq, Show)
 
```

Comment:
It would be good to define a Monad for this at this point - otherwise the more we keep it as functions on IO the harder it will become later on to add other Effects like `Katip` etc. With that said - worth refactoring to a `Reader CompileOpts`

Reply:
Hmm, I don't think it's worth it as this shouldn't grow much more. I'm comfie with keeping this bare for now and we can ofc refactor it into a monad later on if necessary.